onnxruntime/core/graph/contrib_ops/quantization_defs.cc

```diff
@@ -1240,5 +1240,308 @@ ONNX_MS_OPERATOR_SET_SCHEMA(
         .TypeConstraint("F", {"tensor(float16)"}, "Be compatible with float version.")
         .TypeAndShapeInferenceFunction(ONNX_NAMESPACE::propagateShapeAndTypeFromFirstInput));
 
+void convTransposeShapeInference(InferenceContext& ctx,
```

Comment:
Where did all this new shape inferencing logic come from? If it was copied from an ONNX schema can we just call that and augment as needed with checks for the additional inputs instead of cut-and-pasting?

Are there unit tests that exercise all the various paths here? If we call the ONNX shape inferencing we benefit from all the existing testing of that code.

Reply:
Good point.

I can do it by a new InferenceContext